server/src/main/java/com/hedera/block/server/BlockStreamService.java

```diff
@@ -129,6 +138,24 @@ private StreamObserver<BlockStreamServiceGrpcProto.BlockResponse> streamSource(f
 
         return streamObserver;
     }
+
+    private void getBlock(BlockStreamServiceGrpcProto.Block block, StreamObserver<BlockStreamServiceGrpcProto.Block> responseObserver) {
+        String message = "GET BLOCK RESPONSE! ";
+        LOGGER.log(System.Logger.Level.INFO, "GetBlock request received");
+        Optional<BlockStreamServiceGrpcProto.Block> responseBlock = blockPersistenceHandler.read(block.getId());
+        if(responseBlock.isPresent()) {
+            LOGGER.log(System.Logger.Level.INFO, "SENDING BLOCK # " + block.getId());
+            complete(responseObserver, responseBlock.get()); // TODO: Should return int and not quoted string
+        } else {
+            LOGGER.log(System.Logger.Level.INFO, "DID NOT FIND YOUR BLOCK");
+            // TODO: Fix below. It could return gRPC equivalent of 404 NOT FOUND
+            responseObserver.onError(Status.NOT_FOUND
+                    .withDescription("DID NOT FIND YOUR BLOCK")
+                    .asRuntimeException()
+            );
+            // complete(responseObserver, BlockStreamServiceGrpcProto.Block.getDefaultInstance());
```

Comment:
do we need this commented code?

Reply:
Keeping this comments for the fix in TODO: in the code.